rust/src/measurements/canonical_noise/mod.rs

```diff
@@ -63,7 +66,8 @@ pub fn make_canonical_noise(
         Approximate(MaxDivergence),
         Function::new_fallible(move |&arg: &f64| {
             let canonical_rv = CanonicalRV {
-                shift: RBig::try_from(arg).unwrap_or(RBig::ZERO),
+                shift: RBig::try_from(arg.clamp(f64::MIN_FINITE, f64::MAX_FINITE))
```

Comment:
I'm surprised it's not plain `MIN` and `MAX`?
https://doc.rust-lang.org/std/primitive.f64.html#associatedconstant.MAX